src/MessageReader.test.js

```diff
@@ -398,5 +396,15 @@ describe("MessageReader", () => {
       // We shouldn't expose constants on the message.
       expect(Object.keys(message).some((key) => key === "STALE")).toBe(false);
     });
+
+    it("freezes the resulting message if requested", () => {
+      const reader = new MessageReader("string firstName \n string lastName\nuint16 age", { freeze: true });
+      const buffer = Buffer.concat([getStringBuffer("foo"), getStringBuffer("bar"), new Buffer([0x05, 0x00])]);
+      const output = reader.readMessage(buffer);
+      expect(output).toEqual({ firstName: "foo", lastName: "bar", age: 5 });
```

Comment:
Please extend the test to show that sub-messages are frozen recursively, and that it extends to time/duration values which are "primitives" and not technically nested messages.

```suggestion
      expect(output).toEqual({ firstName: "foo", lastName: "bar", age: 5 });
      expect(Object.isFrozen(output)).toBe(true);
```

Reply:
Yeah good idea, and also good point of just using `Object.isFrozen` in tests! 😄